app/models/vacation.rb

```diff
@@ -10,9 +11,33 @@ class Vacation < ApplicationRecord
   enumerize :status, in: {
     pending: 0,
     approved: 1,
-    denied: 2
+    denied: 2,
+    cancelled: 3,
   }, predicates: true, scope: :shallow
 
   validates :start_date, :end_date, :user, presence: true
   validates :end_date, comparison: { greater_than: :start_date }
+  validates :start_date, comparison: { greater_than: Time.zone.today }
+
+  def starting_day=(date)
+    @starting_day = date.presence
+    mount_start_date
+    date
+  end
+
+  def ending_day=(date)
+    @ending_day = date.presence
+    mount_end_date
+    date
+  end
```

Comment:
sorry I didn't quite understood what are these methods doing and where they're being used, could you elaborate a little bit on them?

Reply:
I reimplemented in a better way now, it was a mistake